students/shibin_mathew/lesson9/mailroom.py

```diff
@@ -0,0 +1,152 @@
+#!/usr/bin/env python3
+import sys
+from collections import defaultdict
+
+
+class Donor:
+    def __init__(self, names, amount):
+        self.names = names
+        self.amount = amount
+
+    @property
+    def names(self):
+        return "{}".format(self._names)
+
+    @names.setter
+    def names(self, names):
+        self._names = names
+
+    @property
+    def amount(self):
+        return self._amount
+
+    @amount.setter
+    def amount(self, amount):
+        self._amount = amount
+
+
+class Collection:
+    def __init__(self, donors2=None):
+        self.donors2 = donors2
+
+    @property
+    def donors2(self):
+        return self._donors2
+
+    @donors2.setter
+    def donors2(self, donors2):
+        self._donors2 = donors2
+
+    def print_donor_names(self, donors2):
+        list_test = []
+        for names in donors2:
+            list_test.append(names)
+        return list_test
+
+    def send_thank_you(self, donors2):
+        #  Function to Send a Thank You Email
+        c1 = Collection()
+        program_quit = False
+        while not program_quit:
+            name = input("Please input your full name\n")
+            if name == "list":
+                print("list of names")
+                print(c1.print_donor_names(donors2))
+            elif name == 'quit' or name == 'Quit':
+                break
+            else:
+                break
+        while True:
+            donation = input("Please enter a donation amount\n")
+            if donation == 'quit' or donation == 'Quit':
+                return
+            else:
+                try:
+                    donation = float(donation)
+                    donors2[name].append(donation)
+                    break  # break out of the while loop
+                except ValueError:
+                    print("please input a valid amount")
+
+    def create_report(self, donors2):
+        #  Function to create a report
+        list_test = []
+        print("hello")
+        print("{: <20s}{: ^4s}{: <10s}{: ^4s}{: <10s}{: ^4s}{: <10s}"
+              .format('Donor Name', '|', 'Total Given', '|',
+                      'Num Gifts', '|', 'Average Gift'))
+        print('-' * 70)
+        print(' ')
+        for names in donors2:
```

Comment:
the main point of the class is to use the attribute (`self.donors2`), why is this accepting it as a param instead?

Reply:
Good point - this was my mistake. I made the change.